pkg/group/repository.go

```diff
@@ -33,6 +33,21 @@ func (r repository) find(name string) (*model.Group, error) {
 	return group, err
 }
 
+func (r repository) findWithDetails(name string) (*model.Group, error) {
+	var group *model.Group
+	err := r.db.
+		Where("name = ?", name).
+		Joins("ClusterConfiguration").
+		Preload("Users").
+		Preload("AdminUsers").
+		First(&group).Error
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		return nil, errdef.NewNotFound("group %q doesn't exist", name)
+	}
+
+	return group, err
```

Comment:
I know we are not 100% consistent yet but we did say that we were going to return our own errors instead of 3rd party library ones. Here you are returning a gorm error that will then directly get passed to a user https://github.com/dhis2-sre/im-doc/blob/master/dev/README.md#general